Source/Core/Core/PowerPC/CachedInterpreter/CachedInterpreter.h

```diff
@@ -50,6 +50,7 @@ class CachedInterpreter : public JitBase
                                                  UGeckoInstruction data);
   static void WritePC(CachedInterpreter& cached_interpreter, UGeckoInstruction data);
   static void WriteBrokenBlockNPC(CachedInterpreter& cached_interpreter, UGeckoInstruction data);
+  friend void DoBranchWatchHit(CachedInterpreter& cached_interpreter, UGeckoInstruction n);
```

Comment:
Is this intended to be static void, rather than friend void?

Reply:
`CachedInterpreter::DoBranchWatchHit` accesses the protected member, `JitBase::m_code_buffer`.  If this were an inline definition, I could just use static.  However, non-member functions defined outside of the class must use the friend declaration to access private members.
https://en.cppreference.com/w/cpp/language/friend

Reply:
Right, but, why does it need to be a friend function when a static member would do the [same](https://godbolt.org/z/jxrMxj9qf) thing and also be consistent with the rest of the identifiers its around?

Like, unless I'm overlooking something obvious at 3AM, they should be essentially equivalent.

Reply:
Man, I have no clue what the difference between your Compiler Explorer example and the code in CachedInterpreter is, but I get `Error	C2248	'JitBase::m_code_buffer': cannot access protected member declared in class 'JitBase' ` from changing it from friend to static. I thought the info on cppreference sounded cut-and-dry like what I needed.

Reply:
Wait, yes I do.